### PR TITLE
Feature: Cars API Integration

### PR DESCRIPTION
**Pull request by ahmidou09**
This pull request introduces the integration of API calls for managing cars in the application.

## Changes
- Added asynchronous thunks for fetching, adding, and removing cars.
- Utilized Axios for making API requests to `http://localhost:3000/items`.
- Updated Redux state based on asynchronous action status.

### CODE REVIEW

**Review by ahmed025566 — APPROVED** [review]
Nice job, Mahdi

File: src/components/pages/Car/Cars.jsx [hunk 1]
```diff
@@ -1,9 +1,44 @@
...
+
+  useEffect(() => {
+    dispatch(fetchCars());
+  }, [dispatch]);
+
+  return (
+    <div className="cars-container">
+      <h1>List of Cars</h1>
+      {status === 'loading' && <p>Loading...</p>}
+      {status === 'failed' && <p>Error loading cars.</p>}
```
Comment (Armynerh) [inline src/components/pages/Car/Cars.jsx:21]: OPTIONAL CHANGES
On loading the dashboard, as a new user who hasn't created any cars, it displays "Error loading cars." Why not make a condition to tell the user they haven't created any cars? for example, for "You have not created any cars." 

Screenshot 2023-12-23 at 14 25 59

**Review by Armynerh — APPROVED** [review]
APPROVED 
Hi @ahmidou09  ,
Your project is complete! There is nothing else to say other than... it's time to merge it 

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- well-written code
- Correct Linters

Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.

**Review by alexansaa — APPROVED** [review]
Nice Job. Please, go ahead and keep up the great work!

File: src/components/pages/Car/Cars.jsx (hunk 1 above)
Comment (ahmidou09) [reply src/components/pages/Car/Cars.jsx:21]: The message "Error loading cars" typically indicates a technical server issue, not the absence of created cars.

**Pull request merged**